testing/helpers/type_helpers/type_helpers.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable @typescript-eslint/no-type-alias */
+export type PropertyType<T, TProp extends PropertyKey> =
+  T extends (infer T2)[]
+    ? PropertyType<T2, TProp>
+    : T extends Partial<Record<TProp, infer TValue>>
+      ? TValue
+      : never;
+
+export type DeepPropertyType<T, TProps> =
```

Comment:
Consider this renaming:
```suggestion
export type RecursivePropertyType<T, TProps> =
```

Reply:
Done.

Reply:
Also changed tuples to string templates. Think `'a.b.c'` is prettier than ['a', 'b', 'c']